### DISCUSSION TITLE
Add custom HTML to "featured posts" page

### DISCUSSION THREAD

**Discussion by RampantPanda** [GetPublii/Publii#1485] [Help] [answered]
So I'm using the "featured posts" page as my homepage. I need to add custom HTML to the page (to verify the page as mine in Mastodon). How could I do this? 
Thanks.

**Comment by candidexmedia** [comment]
Hi @RampantPanda! Verifying your site on Mastodon is the same regardless of what you use for your homepage, and I've implemented it on [my own site](https://www.candide.media) and [Mastodon profile](https://mastodon.design/@candidexmedia).

To do so, copy the `rel="me"` code that Mastodon provides in your instance's `/settings/verification` page:

![image](https://github.com/GetPublii/Publii/assets/95003834/31493ee9-d795-4a7b-9003-d4b92dc3b0ff)

Then, in Publii, go to Tools & Plugins > Custom HTML > Body:

![image](https://github.com/GetPublii/Publii/assets/95003834/f6fb5e5f-d109-4d36-92b3-776f7c323c23)

Paste the code you copied, adding `style="display:none"` so that the link is invisible.

```html
<!-- Mastodon --->
<a rel="me" href="https://mastodon.instance/@username" style="display:none">Mastodon</a>
```

Save the settings, then Sync your website. Go back to your Mastodon Settings Verification page to check whether it was successful (should be pretty quick). If you run into any issues, make sure that the link in your profile matches your site's URL exactly and [remove any trailing slashes](https://www.reddit.com/r/Mastodon/comments/1bojri7/comment/kwpi5st/?context=3).

The nice thing about this technique is that it will add the verification to every page on your site, which is great for discoverability on plugins like [StreetPass for Mastodon](https://streetpass.social/) or for automatically verifying any other link to a page on your site.

If you want the verification to be tied to an existing Mastodon hyperlink/hyperlinked logo image, simply add `rel="me"` to the `<a>` tag.

**Reply by candidexmedia** [reply, marked as answer]
I've [just learned](https://github.com/mastodon/mastodon/issues/10345#issuecomment-475682607) that it may also be possible to create the verification using a `<link>` tag in the `<head>`, which doesn't involve having to add inline CSS to hide the link.

To do so, go to Tools & Plugins > Custom HTML > Head, and paste the following (changing the `href` value to your profile's link):

```
<!-- Mastodon Verification --->
<link rel="me" href="https://mastodon.instance/@username"/>
```

**Reply by candidexmedia** [reply]
Just wanted to share an update: if your Mastodon instance is running the latest version, it's now possible to have an author attribution byline added to link previews for posts coming from your site, like this:

![](https://i.imgur.com/H88ke2Z.png)

Instructions: https://rknight.me/blog/setting-up-mastodon-author-tags/
